clippy_utils/src/lib.rs

```diff
@@ -1846,6 +1846,14 @@ pub fn is_lint_allowed(cx: &LateContext<'_>, lint: &'static Lint, id: HirId) ->
     cx.tcx.lint_level_at_node(lint, id).0 == Level::Allow
 }
 
+/// Returns `true` if any of the `impl`s for the given `item` has the lint allowed
+pub fn any_impl_has_lint_allowed(cx: &LateContext<'_>, lint: &'static Lint, id: DefId) -> bool {
+    cx.tcx
+        .inherent_impls(id)
```

Comment:
Sorry, a bit late, but does this also return *trait* impls? `tcx.inherent_impls` only contains those impls with no trait reference, right? If it does work correctly, I think it should at least have a test, since that's usually what derive macros expand to and AIUI what the issue asked for ?

Though reading https://github.com/rust-lang/rust-clippy/issues/12035#issuecomment-1906822877, it doesn't seem uncontroversial that serde should annotate its impls with this anyway and maybe deserves more discussion in general?

Reply:
oh yeah good point.

perhaps we should wait on this PR for now

Reply:
also looking up _all_ trait impls can be an expensive operation